packages/typescript/test/test.js

```diff
@@ -1129,3 +1130,68 @@ function fakeTypescript(custom) {
     custom
   );
 }
+
+test.serial('picks up on newly included typescript files in watch mode', async (t) => {
+  // clean up artefacts from earlier builds
+  const fileNames = fs.readdirSync('fixtures/watch');
+  fileNames.forEach((fileName) => {
+    if (path.extname(fileName) === '.ts') {
+      fs.unlinkSync(path.join('fixtures/watch', fileName));
+    }
+  });
+
+  // set up initial main.ts
+  // (file will be modified later in the test)
+  fs.copyFileSync('fixtures/watch/main.ts.1', 'fixtures/watch/main.ts');
+
+  const watcher = watch({
+    input: 'fixtures/watch/main.ts',
+    output: {
+      dir: 'fixtures/watch/dist'
+    },
+    plugins: [
+      typescript({
+        tsconfig: 'fixtures/watch/tsconfig.json',
+        target: 'es5'
+      })
+    ],
+    watch: {
+      chokidar: {
+        // make sure watch mode works in Windows CI; https://github.com/paulmillr/chokidar/issues/998
+        usePolling: true,
+        interval: 10
+      }
+    },
+    onwarn
+  });
+
+  await waitForWatcherEvent(watcher, 'END');
+
+  // add new .ts file
+  fs.copyFileSync('fixtures/watch/new.ts.1', 'fixtures/watch/new.ts');
+
+  // update main.ts file to include new.ts
+  fs.copyFileSync('fixtures/watch/main.ts.2', 'fixtures/watch/main.ts');
+
+  await waitForWatcherEvent(watcher, 'END');
+
+  watcher.close();
+
+  const code = fs.readFileSync('fixtures/watch/dist/main.js');
+  const usage = code.includes('Is it me');
+  t.true(usage, 'should contain usage');
+});
+
+function waitForWatcherEvent(watcher, eventCode) {
+  return new Promise((resolve, reject) => {
+    watcher.on('event', function handleEvent(event) {
```

Comment:
You can remove the `off` calls with `once`:
```suggestion
    watcher.once('event', function handleEvent(event) {
```

Reply:
Not in this case, unfortunately. :slightly_smiling_face: 

`once` removes the listener after the first event of a type is triggered, but this function doesn't listen for the first event of a type. Rollup issues all watcher events as events of type `event` -- if I used `once` here then it would remove the listener directly after the initial (`BUNDLE` / `BUNDLE_START`) events.